repos/system_upgrade/el8toel9/actors/fixgrubefiwrapper/actor.py

```diff
@@ -0,0 +1,34 @@
+import os
+
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.stdlib import api, run
+from leapp.models import FirmwareFacts, TransactionCompleted
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, RPMUpgradePhaseTag
+
+
+class FixGrubEFIWrapper(Actor):
+    """
+    Update /boot/efi/EFI/redhat/grub.cfg on UEFI systems.
+
+    See [RHEL-36186](https://issues.redhat.com/browse/RHEL-36186).
+    """
+
+    name = 'fix_grub_efi_wrapper'
+    consumes = (TransactionCompleted, FirmwareFacts)
+    produces = (Report,)
+    tags = (RPMUpgradePhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        ff = next(self.consume(FirmwareFacts), None)
+        if not ff:
+            raise StopActorExecution(
+                'Could not identify system firmware',
+                details={'details': 'Actor did not receive FirmwareFacts message.'}
+            )
```

Comment:
`StopActorExecution` does not accept any msg / parameter. I suggest to log warning msg and just `return`. The `StopActorExecution` exception is used just to stop the execution, so `return` should be equal.

Reply:
OK, will push an updated patch soon